cli/run_util_test.go

```diff
@@ -515,3 +519,138 @@ workflows:
+
+	testDir, err := initializeStepDir(&additionalEnvironments)
+	if err != nil {
+		t.Fatalf("failed to create test dir, error: %s", err)
+	}
+
+	if filepath.Dir(testDir) != tempTestResultsDir {
+		t.Fatal("BITRISE_TEST_RESULT_DIR should be a child of BITRISE_TEST_RESULTS_DIR")
+	}
+
+	if len(additionalEnvironments) != 1 {
+		t.Fatal("should be only one additionalEnvironments")
+	} else {
+		key, value, err := additionalEnvironments[0].GetKeyValuePair()
+		if err != nil {
+			t.Fatalf("failed to get GetKeyValuePair, error: %s", err)
+		}
+		if key != "BITRISE_TEST_RESULT_DIR" {
+			t.Fatal("key should be BITRISE_TEST_RESULT_DIR")
+		}
+		if value != testDir {
+			t.Fatal("value should be the generated test dir path")
+		}
+	}
+
+	if exists, err := pathutil.IsDirExists(testDir); err != nil {
+		t.Fatalf("failed to check if dir exists, error: %s", err)
+	} else if !exists {
+		t.Fatal("BITRISE_TEST_RESULT_DIR path should exists on the FS")
+	}
+}
+
+func TestNormalizeStepDir(t *testing.T) {
+	t.Log("test empty dir")
+	{
+		testDirPath, err := pathutil.NormalizedOSTempDirPath("testing")
+		if err != nil {
+			t.Fatalf("failed to create testing dir, error: %s", err)
+		}
+
+		testResultStepInfo := models.TestResultStepInfo{}
+
+		exists, err := pathutil.IsDirExists(testDirPath)
+		if err != nil {
+			t.Fatalf("failed to check if dir exists, error: %s", err)
+		}
+
+		if !exists {
+			t.Fatal("test dir should exits")
+		}
+
+		if err := normalizeTestDir(testDirPath, testResultStepInfo); err != nil {
+			t.Fatalf("failed to normalize test dir, error: %s", err)
+		}
+
+		exists, err = pathutil.IsDirExists(testDirPath)
+		if err != nil {
+			t.Fatalf("failed to check if dir exists, error: %s", err)
+		}
+
+		if exists {
+			t.Fatal("test dir should not exits")
+		}
+	}
+
+	t.Log("test not empty dir")
+	{
+		testDirPath, err := pathutil.NormalizedOSTempDirPath("testing")
+		if err != nil {
+			t.Fatalf("failed to create testing dir, error: %s", err)
+		}
+
+		testResultStepInfo := models.TestResultStepInfo{}
+
+		exists, err := pathutil.IsDirExists(testDirPath)
+		if err != nil {
+			t.Fatalf("failed to check if dir exists, error: %s", err)
+		}
+
+		if !exists {
+			t.Fatal("test dir should exits")
+		}
+
+		if err := fileutil.WriteStringToFile(filepath.Join(testDirPath, "test-file"), "test-content"); err != nil {
+			t.Fatalf("failed to write file, error: %s", err)
+		}
+
+		if err := normalizeTestDir(testDirPath, testResultStepInfo); err != nil {
+			t.Fatalf("failed to normalize test dir, error: %s", err)
+		}
+
+		exists, err = pathutil.IsDirExists(testDirPath)
+		if err != nil {
+			t.Fatalf("failed to check if dir exists, error: %s", err)
+		}
+
+		if !exists {
+			t.Fatal("test dir should exits")
+		}
+
+		exists, err = pathutil.IsPathExists(filepath.Join(testDirPath, "test-file"))
+		if err != nil {
+			t.Fatalf("failed to check if dir exists, error: %s", err)
+		}
+
+		if !exists {
+			t.Fatal("test file should exits")
```

Comment:
Similar as above. `test-file` is created implicitly by `fileutil.WriteStringToFile`. If test fails with this error, it is not immediately obvious what went wrong.